compiler/src/dotty/tools/dotc/core/Names.scala

```diff
@@ -549,7 +549,8 @@ object Names {
   private def hashValue(cs: Array[Char], offset: Int, len: Int): Int = {
     var i = offset
     var hash = 0
-    while (i < len + offset) {
+    val until = offset + len
```

Comment:
I doubt this has any effect. Hoisting constant expressions out of loops is among the easiest optimizations of a JIT compiler. So I think it's better to revert this commit. 

Otherwise LGTM.